engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/ordering/SortVectorNode.java

```diff
@@ -0,0 +1,836 @@
+    return Boolean.TRUE.equals(object);
+  }
+
+  /** Returns true iff the given array of comparators is all Default_Comparator */
+  boolean areAllDefaultComparators(InteropLibrary interop, Object comparators) {
+    assert interop.hasArrayElements(comparators);
+    var ctx = EnsoContext.get(this);
+    try {
+      int compSize = (int) interop.getArraySize(comparators);
+      for (int i = 0; i < compSize; i++) {
+        assert interop.isArrayElementReadable(comparators, i);
+        Object comparator = interop.readArrayElement(comparators, i);
+        if (!isDefaultComparator(comparator, ctx)) {
+          return false;
+        }
+      }
+    } catch (UnsupportedMessageException | InvalidArrayIndexException e) {
+      throw new IllegalStateException("Should not be reachable", e);
+    }
+    return true;
+  }
+
+  boolean isDefaultComparator(Object object, EnsoContext ctx) {
+    return ctx.getBuiltins().defaultComparator().getType() == object;
+  }
+
+  private boolean isNan(Object object) {
+    return object instanceof Double dbl && dbl.isNaN();
+  }
+
+  private enum ProblemBehavior {
+    IGNORE,
+    REPORT_WARNING,
+    REPORT_ERROR;
+
+    static ProblemBehavior fromInt(int intValue) {
+      return switch (intValue) {
+        case 0 -> IGNORE;
+        case 1 -> REPORT_WARNING;
+        case 2 -> REPORT_ERROR;
+        default -> throw new IllegalArgumentException(Integer.toString(intValue));
+      };
+    }
+  }
+
+  /**
+   * Group of elements grouped by comparator.
+   *
+   * @param elems Elements of the group.
+   * @param comparator SortComparator for the elems, i.e., it should hold that {@code elems.each
+   *     it-> (Comparable.from it) == comparator}.
+   * @param compareFunc `SortComparator.compare` function extracted from the comparator.
+   */
+  private record Group(List<Object> elems, Type comparator, Function compareFunc) {}
+
+  /**
+   * Convenient base class that implements java.util.Comparator, and gathers warnings about
+   * incomparable values. The warnings are gathered as pure Strings in a hash set, so that they are
+   * not duplicated.
+   */
+  private abstract static class SortComparator implements java.util.Comparator<Object> {
+
+    private final Set<String> warnings = new HashSet<>();
+    final AnyToTextNode toTextNode;
+    final ProblemBehavior problemBehavior;
+    final InteropLibrary interop;
+
+    protected SortComparator(
+        AnyToTextNode toTextNode, ProblemBehavior problemBehavior, InteropLibrary interop) {
+      this.toTextNode = toTextNode;
+      this.problemBehavior = problemBehavior;
+      this.interop = interop;
+    }
+
+    @TruffleBoundary
+    protected void attachIncomparableValuesWarning(Object x, Object y) {
+      var xStr = toTextNode.execute(x).toString();
+      var yStr = toTextNode.execute(y).toString();
+      String warnText = "Values " + xStr + " and " + yStr + " are incomparable";
+      warnings.add(warnText);
+    }
```

Comment:
We will be adding a warning for each pair of incomparable values when sorting.

If I sort a vector of 1M of values, each of them being incomparable, how many of these warnings will I get?

I'm thinking that we may want to report only the first 3-10 of these and then report just a counter _'and N more values that were incomparable.'_

See for example `Additional_Warnings` in the Table library which is a similar concept. When doing Table aggregations we report the first 10 issues and then just report the count of additional problems. We don't want 1000+ of the same problem just with different values in it attached, it will not add much more information.

More importantly, sorting 1M elements that are comparable vs incomparable shouldn't have vastly different performance characteristics.

I'm not pushing to necessarily do it _now_, but ideally we should at least create a ticket to ensure that large incomparable vectors are handled well by this. Ideally with adding a test that sorts a vector of at least 10k+ values and measures the amount of warnings, possibly also a benchmark.

Reply:
Tracked in #6283, as a follow-up.